daemon/info_unix.go

```diff
@@ -57,29 +58,64 @@ func (daemon *Daemon) fillPlatformInfo(v *system.Info, sysInfo *sysinfo.SysInfo,
 	v.ContainerdCommit.ID = "N/A"
 	v.InitCommit.ID = "N/A"
 
-	if _, _, commit, err := parseDefaultRuntimeVersion(&cfg.Runtimes); err != nil {
-		log.G(context.TODO()).Warnf(err.Error())
-	} else {
+	if err := func() error {
+		_, _, commit, err := parseDefaultRuntimeVersion(&cfg.Runtimes)
+		if err != nil {
+			log.G(ctx).WithError(err).Warn("Failed to retrieve default runtime version")
+			return nil
+		}
 		v.RuncCommit.ID = commit
+		return nil
+	}(); err != nil {
+		return err
```

Comment:
Looks like this now never returns an error, so we may be able to get rid of the closure again? (at least until we have something in it that handles context-cancellation, but we currently don't)

Reply:
Fixed.